components/layout_2020/quotes.rs

```diff
@@ -0,0 +1,1291 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+// Quotes data is obtained from ICU CLDR data file /tmp/cldr-common-46.0.zip.
+
+// TODO(xiaochengh): This file should better be moved to elsewhere and maintained automatically.
+// Or even better, extend the icu create to provide the data directly.
+
+use std::collections::HashMap;
+use std::sync::OnceLock;
+
+use icu_locid::Locale;
+
+#[derive(Clone, Copy, Debug)]
+struct QuotesData(char, char, char, char);
+
+#[derive(Clone, Copy, Debug)]
+pub struct QuotePair {
+    pub opening: char,
+    pub closing: char,
+}
+
+type QuotesMap = HashMap<&'static str, QuotesData>;
+
+static QUOTES_MAP: OnceLock<QuotesMap> = OnceLock::new();
+
+static DEFAULT_QUOTES: QuotesData = QuotesData('\u{201c}', '\u{201d}', '\u{2018}', '\u{2019}');
+
+fn create_quotes_map() -> QuotesMap {
+    let input = [
+        (
+            "aa",
+            QuotesData('\u{201c}', '\u{201d}', '\u{2018}', '\u{2019}'),
```

Comment:
Hrm. What I was imagining here was that this HashMap would essentially be `HashMap<&'static str, QuotePair>;` and you would convert all of the QuotesData entries in the initialization to QuotePair. I think it makes more sense to do the conversion ahead of time rather than at runtime.

Reply:
Done